Plans: Show fewer features in the grid

oswian opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to [#3332](https://github.com/Automattic/martech/issues/3332) 

## Proposed Changes

Updates the features grid on `/start/plans` and `/plans/:site` to fix inaccuracies, and to display fewer features per plan so that differentiating features are visible above the fold. Further context in the project thread (p5uIfZ-fIN-p2).

This experiment includes two treatment variants:
  
- Treatment A (focuses on fixing inaccuracies)
  - Some features added
  - Many features removed
  - Updated titles and descriptions for various features
  - Ordering of features has changed
  
- Treatment B (extends Treatment A, focuses on displaying less features)
  - More features removed
  - Vertical alignment of all features
  - Removed the copy: "Everything in xx, plus"
  - Larger font size for features
 
For details on how we are measuring the impact of these changes, please review the experiment post (pbxNRc-45y-p2).

---

**Treatment A**

| Control | Treatment |
| -- | -- |
|<img size=400 src="https://github.com/user-attachments/assets/e14098d0-33a3-487b-8b87-db2faa47974d" />|<img size=400 src="https://github.com/user-attachments/assets/0a8953a8-21ed-4f03-8d23-d6a15cecd156" />|

---

**Treatment B**

| Control | Treatment |
| -- | -- |
|<img size=400 src="https://github.com/user-attachments/assets/e2c6f299-40e5-4dcf-8820-20f9d0250725" />|<img size=400 src="https://github.com/user-attachments/assets/bb34c2d8-7950-411d-89f8-d626d6423be9" />|



## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

The hypothesis is that a simplified features grid, displaying differentiators above the fold (pbxNRc-3Tw-p2), will help users find the right plan and have a positive impact on sign-ups and cash sales. 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Please verify the following:
  - The treatment variants appear as per the above screenshots
  - The changes are limited to `/start/plans` and `/plans/:site`
  - The control variant appears as per production (ie. no changes)

Treatment A can be tested with the flag `simplified-features-grid-a`:
* `/start/plans?flags=simplified-features-grid-a`
* `/plans/:site?flags=simplified-features-grid-a`

Treatment B can be tested with the flag `simplified-features-grid-b`:
* `/start/plans?flags=simplified-features-grid-b`
* `/plans/:site?flags=simplified-features-grid-b`

---

Stories have also been added for each treatment variant:

- 'Experiment [Simplified features grid]: Treatment A'
- 'Experiment [Simplified features grid]: Treatment B'

<img size=400 src="https://github.com/user-attachments/assets/d2a2f33b-39b5-4f9b-8376-b642fcd312cd" />





## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

oswian committed:
Show fewer differentiating features per plan

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16621211

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @oswian for including a screenshot in the description! This is really helpful for our translators.

chriskmnds commented on packages/plans-grid-next/src/components/features-grid/plan-features-list.tsx:
```diff
@@ -132,7 +134,7 @@ const PlanFeaturesList = ( {
 					className="plan-features-2023-grid__table-item"
 				>
-					{ featureGroup?.getTitle() && (
+					{ ! enableCategorisedFeatures && featureGroup?.getTitle() && (
```
I don't quite follow this change. Do you mind explaining? If a title exists, then it should be rendered - that's the idea throughout here. 

It's also quite an obscured/cryptic condition semantically - "if we don't want categorised features and have a group/category title"?

oswian replied:
Good catch, thanks. We don't need the additional check, `getTitle` will handle this - I've updated the feature group title to be `null`. 

[5e16d3a](https://github.com/Automattic/wp-calypso/pull/94044/commits/5e16d3a2f63a2e3caf0c64e6e6c117bfa261083e)

chriskmnds reviewed (changes requested):
Looks beautiful. Left a comment below.

emilyaudela commented on packages/calypso-products/src/features-list.tsx:
```diff
@@ -2284,7 +2303,10 @@ const FEATURES_LIST: FeatureList = {
-		getTitle: () => i18n.translate( 'Optimized WooCommerce hosting' ),
+		getTitle: () =>
+			isAssignedToFewerFeaturesExperiment()
+				? i18n.translate( 'eCommerice tools, integrations, and optimized WooCommerce hosting' )
```
Small typo: eCommerice --> eCommerce

oswian replied:
Thanks, typo fixed.

[5493ac1](https://github.com/Automattic/wp-calypso/pull/94044/commits/5493ac155540855175f4354b7ef2dca947148ac7)

oswian committed:
Use existing check for category title text to control its visibility

oswian committed:
Display storage feature with description

oswian committed:
Fix typo in feature title

aneeshd16 commented on client/my-sites/plans-features-main/hooks/use-fewer-features-experiment.ts:
The linked experiment (21954-explat-experiment) contains a single treatment variant. It might already be on your TODO list to modify the experiment so that it contains both treatment variants, but noting it here just in case. :)

oswian replied:
Thanks for checking, yes this is on my todo's  👍

oswian commented:
I've removed the "String Freeze" label as the experiment is now EN-only (discussion: p5uIfZ-fIJ-p2).

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16621211).

oswian committed:
Add new features and copy updates

oswian committed:
Use larger font size for feature titles

oswian committed:
Allow storage feature to be displayed as badge or text

oswian committed:
Disable feature grouping for second treatment

oswian commented:
For anyone following along -- there are ongoing discussions about what changes should be included in the treatments (see: p5uIfZ-fIJ-p2) so for now I've removed the "Needs Review" label and moved this back to "In Progress". Once the treatment changes are decided on and implemented, I'll re-request a review.

oswian committed:
Update to use experiment profile variations

oswian committed:
Update features for the 'fix inaccuracies' treatment variant

chriskmnds commented on config/development.json:
you don't need these config properties if they are all false. you can enable/disable flags in URL `?flags=-flag-to-disable`

oswian committed:
Update feature config for treatment variants

oswian commented:
<img width=400 src="https://github.com/user-attachments/assets/c203d626-f14b-410d-a03b-9b1fe5b78388" />

😄 Matticbot is quite keen for this PR to have a 'Needs Review' label (after adding 'In Progress' label and converting back to draft to make changes).

oswian commented:
Hi @chriskmnds, I noticed you are AFK next week and have requested changes here. I addressed your review [comment](https://github.com/Automattic/wp-calypso/pull/94044#discussion_r1740318890), can the 'changes requested' now be cleared? Please let me know if there's anything else.

aneeshd16 commented on packages/calypso-products/src/features-list.tsx:
```diff
@@ -2406,24 +2492,41 @@ const FEATURES_LIST: FeatureList = {
-				: i18n.translate( 'Expert support' ),
+		getTitle: () => {
+			if ( isAssignedToSimplifiedFeaturesGridExperimentVariant( 'simplified' ) ) {
+				return i18n.translate( 'Support access to our expert team' );
```
I'm not a native speaker, but it sounds awkward to me. 🤔 Maybe something like "Support from our expert team" sounds better?

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/26d88bc2-d2ff-4169-9c45-766b71ecad47">

chriskmnds replied:
This reads a little weird to me too. It reads like "We support access to our expert team"

oswian replied:
I agree. I'll raise it with the team.

oswian replied:
Updated to: "Fast support from our expert team"

aneeshd16 reviewed (commented):
**Tested**

- [x] The treatment variants appear as per the above screenshots
- [x] The changes are limited to /start/plans and /plans/:site
- [x] The control variant appears as per production (ie. no changes)

I haven't gone through the code yet, will do soon! I have some additional feedback to share:

1. In the comparison grid, I believe this feature should be replaced with "Unlimited automatic shares in social media" to match the feature shown on the features grid. This is for both treatment variants.
<img width="1457" alt="image" src="https://github.com/user-attachments/assets/7a4b98c7-e16b-4d72-8a62-727f09d79a89">

2. On mobile, the free plan card has a strange feel to it. The copy says "Show all features", but clicking on it only shows one feature:
<img width="488" alt="image" src="https://github.com/user-attachments/assets/03f2133f-c677-4daa-abd3-704eba57c4b2">

3. On mobile, I think we can preserve the font size and spacing of the current production design. Otherwise, the text seems unnecessarily spaced out:
<img width="510" alt="image" src="https://github.com/user-attachments/assets/6f541eeb-1b77-4ab2-b582-5ca777b033f7">

4. I'm not sure where to add this design feedback, so adding it here. For the `simplified-features-grid-b`, what if we add check marks to the feature list? It is a common practice to add these to pricing grids, and looks more familiar to the user. WDYT?
<img width="1239" alt="image" src="https://github.com/user-attachments/assets/da6ce3af-d970-4650-aaab-352fcbb70e39">

chriskmnds commented on packages/plans-grid-next/src/components/features-grid/table.tsx:
```diff
@@ -153,6 +154,7 @@ const Table = ( {
 					<tr
 						className={ clsx( 'plans-grid-next-features-grid__feature-group-row', {
 							'is-first-feature-group-row': featureGroupIndex === 0,
+							'is-single-feature-group-row': enableCategorisedFeatures,
```
This feels a bit ambiguous to me. I can't make up what it's referring to and why "enable categorized features" equates to "a single feature group row"

oswian replied:
Good call, I've updated the naming: [4592cd2](https://github.com/Automattic/wp-calypso/pull/94044/commits/4592cd266be720650b7dfaaf41d07e871605ffab)

chriskmnds commented on packages/plans-grid-next/src/types.ts:
```diff
@@ -197,11 +197,9 @@ export type GridContextProps = {
-	/**
-	 * `enableCategorisedFeatures` is no longer exact, and probably best to rename.
-	 * It is only used for showing "Everything in [previous] plus".
-	 */
```
What's changed here? So this is "exact" now?

oswian replied:
I removed this comment as I've introduced a new prop to control "Everything in [previous] plus" (`enablePreviousFeaturesIncludedTitle`), and this experiment utilises `enableCategorisedFeatures` for feature categorisation, so the name seems reasonable? Happy to change it if we can think of a better name.

chriskmnds replied:
>  I've introduced a new prop to control "Everything in [previous] plus" (enablePreviousFeaturesIncludedTitle),

Was this refactor **necessary** to run the experiment? This functionality ("Everything in [previous] plus") is slated for a refactor along with calypso-products to feed the features grid with all the features per plan, not just the unique ones. At that point, the property will be something like "showUniqueFeaturesPerPlan" and that will encompass this more granular property (make it redundant). 

If this is not necessary, then I suggest we move this back to the way it was before and leave for consideration when the more holistic refactor happens. I fear that elevating it into another "acceptable" parameter in the grid will merely put a punt on the proper refactor that needs to happen.

> this experiment utilises enableCategorisedFeatures for feature categorisation, so the name seems reasonable

I'm also not sure about that. All the features are categorised now - there is no "non-categorised features" concept in the grid. The only question is whether to provide titles along with the categories or not. If things linger this way, then let's please replace the comment with details on how it is used for "feature categorization".

chriskmnds commented on packages/plans-grid-next/src/types.ts:
```diff
@@ -197,11 +197,9 @@ export type GridContextProps = {
-	 */
 	enableCategorisedFeatures?: boolean;
+	enableLargeFeatureTitles?: boolean;
+	enableStorageAsBadge?: boolean;
```
NIT. If we can add a brief comment here briefly describing what it refers to, I think it would be great. "storage" feels like a general holder of content in the grid. Something like "display the storage limit/capacity of the plan as a badge like "50GB" or plain text like "50GB storage".

oswian replied:
👍 Prop description added: [f29f7ba](https://github.com/Automattic/wp-calypso/pull/94044/commits/f29f7ba3fe86aa119ed1aa6b3ec9c814532c97b9)

chriskmnds commented on packages/plans-grid-next/src/_shared.scss:
```diff
@@ -75,6 +75,14 @@ $mobile-card-max-width: 440px;
+	padding: 4px 0;
+	font-size: $font-body-small;
+	font-weight: 400;
+	line-height: 20px;
```
All of these, except for `font-size`, are shared with `.plans-grid-next-storage-feature-label__volume-badge`. Should we consolidate?

oswian replied:
Yes good idea, updated: [ebaa5b1](https://github.com/Automattic/wp-calypso/pull/94044/commits/ebaa5b1814f026b565963d6e01952bd0f54963ff)

chriskmnds commented on packages/calypso-products/src/features-list.tsx:
```diff
@@ -516,18 +535,27 @@ const FEATURES_LIST: FeatureList = {
-		getTitle: () => i18n.translate( 'Access to dozens of premium themes ' ),
+		getTitle: () =>
+			isAssignedToSimplifiedFeaturesGridExperiment()
+				? i18n.translate( 'Dozens of premium themes ' )
```
Is the whitespace at the end necessary?

oswian replied:
Thanks, removed.

chriskmnds commented on packages/calypso-products/src/features-list.tsx:
```diff
@@ -516,18 +535,27 @@ const FEATURES_LIST: FeatureList = {
 	},
+	[ FEATURE_THEMES_PREMIUM_AND_STORE ]: {
+		getSlug: () => FEATURE_THEMES_PREMIUM_AND_STORE,
+		getTitle: () => i18n.translate( 'All premium and store themes ' ),
```
Same

oswian replied:
Also removed.

chriskmnds commented on packages/calypso-products/src/features-list.tsx:
```diff
@@ -2604,6 +2720,97 @@ const FEATURES_LIST: FeatureList = {
+	},
+	[ FEATURE_WOO_TAX_SOLUTIONS ]: {
+		getSlug: () => FEATURE_WOO_TAX_SOLUTIONS,
+		getTitle: () => i18n.translate( 'Tax solutions - Avalara & EU Vat' ),
```
Shouldn't `Vat` be `VAT`?

oswian replied:
Yes I think so -- updated.

chriskmnds reviewed (changes requested):
The code overall looks good. Left some comments below. 

One problem I notice is visiting `/plans/:site?flags=simplified-features-grid-b` with a Newsletter site and getting the treatment variation. Is this expected for Newsletter? These sites render the same customized grid/features in admin that you see in `/setup/newsletter/plans`. 

<img width="600" alt="Screenshot 2024-09-06 at 12 16 14 PM" src="https://github.com/user-attachments/assets/ddcb667c-0d9d-41d6-a5fe-16ba595872fa">

What I expected (no change, irrespective of assignment):

<img width="600" alt="Screenshot 2024-09-06 at 12 16 39 PM" src="https://github.com/user-attachments/assets/33c53e5f-70cf-4327-9942-6240c70b40f4">

worth cc'ing @simison

simison commented:
Let's not apply this to tailored plans grids as result looks very broken, like @chriskmnds [noted](https://github.com/Automattic/wp-calypso/pull/94044#pullrequestreview-2282792113).

oswian committed:
Use consistent copy for style feature

oswian committed:
Move dev tools into separate feature group

oswian committed:
Latest treatment feature updates

oswian committed:
Remove redundant config properties

oswian committed:
Feature title copy fixes

oswian committed:
Add comments for new context props

oswian committed:
Clarify prop to control feature group spacing

oswian committed:
Add new prop to control visibility of 'previous plan plus' title

oswian committed:
Consolidate storage label styling

oswian committed:
Remove feature group spacing on mobile

oswian committed:
Hide features for enterprise/vip plan

oswian commented:
Thanks for the review @aneeshd16 

> In the comparison grid, I believe this feature should be replaced with "Unlimited automatic shares in social media" to match the feature shown on the features grid. This is for both treatment variants.

That sounds like a nice simplification. Just to confirm, are you suggesting we change the title in the left-hand column to "Unlimited automatic shares in social media", and place checkmarks where we currently display "Unlimited shares"?

> On mobile, the free plan card has a strange feel to it. The copy says "Show all features", but clicking on it only shows one feature

Good catch. Do you think we should hide the button and expand by default for this treatment? 🤔

> On mobile, I think we can preserve the font size and spacing of the current production design. Otherwise, the text seems unnecessarily spaced out:

Good idea - updated.

> I'm not sure where to add this design feedback, so adding it here. For the simplified-features-grid-b, what if we add check marks to the feature list? It is a common practice to add these to pricing grids, and looks more familiar to the user. WDYT?

I really like this idea. At the moment the features seem to 'float' without an anchor/bullet. It's a fairly significant visual change though, so I'm wondering if it might be better to test as a followup experiment. 🤔 Either way I think it's worth proposing on the feedback requested post: p5uIfZ-fIJ-p2.

aneeshd16 commented:
> That sounds like a nice simplification. Just to confirm, are you suggesting we change the title in the left-hand column to "Unlimited automatic shares in social media", and place checkmarks where we currently display "Unlimited shares"?

Yes! And, the checkmarks should be present for all plans since this feature is available for all plans. I brought this up because, in the feature grid for treatment a, this feature is mentioned in the free plan's feature list and currently the features grid and comparison grid would have a mismatch wrt this feature.
<img width="386" alt="image" src="https://github.com/user-attachments/assets/aa95b0f1-e987-4ebd-a0aa-05c8faad47cd">

> Good catch. Do you think we should hide the button and expand by default for this treatment? 🤔

TBH, I'm not quite sure. The copy would still say, "Show all features," and there would be a solitary feature in the list. Another idea—since there are fewer features in this variant, how about removing the "Show all features" toggle and just showing the full list instead?

> I really like this idea. At the moment the features seem to 'float' without an anchor/bullet. It's a fairly significant visual change though, so I'm wondering if it might be better to test as a followup experiment. 🤔 Either way I think it's worth proposing on the feedback requested post: p5uIfZ-fIJ-p2.

Will do!

oswian committed:
Exclude tailored plans grids from the experiment

oswian commented:
> One problem I notice is visiting /plans/:site?flags=simplified-features-grid-b with a Newsletter site and getting the treatment variation. Is this expected for Newsletter? These sites render the same customized grid/features in admin that you see in /setup/newsletter/plans.

@chriskmnds Thanks for catching this. This experiment should only apply to the standard/regular features grid as shown in the description screenshots (do we have a term we use to refer to the 'standard' version?).

I've updated [the hook](https://github.com/Automattic/wp-calypso/pull/94044/files#diff-817502ee27ce02f480a9e1f9f21467236a3184aea7c050517e360b82e9ff2a5e) to restrict the experiment eligibility to either:

- Within signup with the flow `onboarding`
- Within admin with the intent `plans-default-wpcom`

Does this approach look ok?

Here's what it looks like for `/setup/newsletter/plans` (with either treatment flag applied):

![CleanShot 2024-09-10 at 14 27 42@2x](https://github.com/user-attachments/assets/c2a240ba-2193-4ae2-957a-27337bc3453e)

And for a newsletter site in admin (with either treatment flag applied):

![CleanShot 2024-09-10 at 14 26 16@2x](https://github.com/user-attachments/assets/552bce71-2e19-4bb0-a269-4b3991ff902c)

oswian committed:
Hide storage feature group title

oswian committed:
Re-use existing features definitions

oswian commented:
> Yes! And, the checkmarks should be present for all plans since this feature is available for all plans. I brought this up because, in the feature grid for treatment a, this feature is mentioned in the free plan's feature list and currently the features grid and comparison grid would have a mismatch wrt this feature.

@aneeshd16 👍 I've pushed this update -- it now displays checkmarks for each plan.

> TBH, I'm not quite sure. The copy would still say, "Show all features," and there would be a solitary feature in the list. Another idea—since there are fewer features in this variant, how about removing the "Show all features" toggle and just showing the full list instead?

Yep that's what I had in mind too. I've raised it here: p5uIfZ-fIJ-p2#comment-23715.

chriskmnds commented on packages/plans-grid-next/src/types.ts:
```diff
@@ -197,11 +197,37 @@ export type GridContextProps = {
+	 */
+	enablePreviousFeaturesIncludedTitle?: boolean;
+
+	/**
+	 * Reduce the vertical spacing between each feature group
+	 */
+	enableReducedFeatureGroupSpacing?: boolean;
```
I don't like how these `enableFoo`, `enableBar` (etc) props attempt to pattern-ize the idea of adding granular props to control experiment logic. Almost always this is the result of lacking a better underlying design. I don't want to see `enableX` being projected as an acceptable pattern that can span into some Nth dimension (anyone adding anything without due consideration). The vision with plans-grid package is to reduce the need for new/standalone props whenever we want to run an experiment, unless absolutely necessary.

For this case, there could be other options to consider:

- introduce optional render props that consuming end can tailor design. This can stay, irrespective of experiment result. Can literally call them `renderProps` (`interface RenderProps { Feature: ..., etc. }`)
- overwrite the styling at consuming end (plans-features-main) and follow up to deploy the final change as the default grid styling once/if experiment completes successfully
- more?

Have we considered any of the other options?

oswian replied:
> introduce optional render props that consuming end can tailor design. This can stay, irrespective of experiment result. Can literally call them renderProps (interface RenderProps { Feature: ..., etc. })

Just to clarify are you referring to the render props pattern? ie.

```
<FeaturesGrid
  renderFeatureTitles={ () => ... }
  ...
/>
```

Or `renderProps` as a single prop controlling multiple display properties? ie.

```
<FeaturesGrid 
  renderProps={{
    largeFeatureTitles: true,
    ...
  }}
/>
```

chriskmnds replied:
I am referring to the first/pattern, but let's also not make it super inviting to the "granularity effect". Meaning, it should be a holder of general components (as possible). I'd go for a type that looks like:

```
interface Renderers {
    Feature?: React.FunctionComponent< FeatureRendererProps >
}
```

and add that as an optional prop (`renderers`) on the common grid props and context.

p.s. Totally fine to handle this in a follow-up if it's anything more complex than it looks. Not blocking. We can create an issue and follow up on it with some priority. but I feel it should be relatively straightforward? 🤔

oswian replied:
Ok let's tackle this change in a followup. It does sound relatively straightforward, however it would be good to have some time to explore in more detail (outside of this experiment). I've created an issue: https://github.com/Automattic/wp-calypso/issues/94458

chriskmnds reviewed (commented):
Thanks for following up on the comments :-)

> And for a newsletter site in admin (with either treatment flag applied):
> 
> ![CleanShot 2024-09-10 at 14 26 16@2x](https://private-user-images.githubusercontent.com/69198925/365892319-552bce71-2e19-4bb0-a269-4b3991ff902c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NjM0MjcsIm5iZiI6MTcyNTk2MzEyNywicGF0aCI6Ii82OTE5ODkyNS8zNjU4OTIzMTktNTUyYmNlNzEtMmUxOS00YmIwLWEyNjktNGIzOTkxZmY5MDJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDEwMTIwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NDlmZDYwM2RkMjQ2YTQ2NmUwNmYwYTUwNzU3MjM3NGZhYmY0YzQ5YjU5NjllYTBmNjJhY2ZiNWVmZjcwMDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VlBJLVOfqoYhuiSgeTn81qhKqCSedYQHtTe_JM9r_IY)

This may be ok, but want to double-check just in case. What should happen when user clicks on "View all plans" / escape hatch? Should it result in a grid with treatment or resolve to the original/control irrespective of experiment assignment? I think the latter makes sense, but not sure.

oswian commented:
> This may be ok, but want to double-check just in case. What should happen when user clicks on "View all plans" / escape hatch? Should it result in a grid with treatment or resolve to the original/control irrespective of experiment assignment? I think the latter makes sense, but not sure.

@chriskmnds Good question, I've raised it here: p5uIfZ-fIJ-p2#comment-23848

oswian committed:
Revert to existing prop to control feature category titles

oswian commented:
> All the features are categorised now - there is no "non-categorised features" concept in the grid. 

@chriskmnds Thanks for clarifying. The `enableCategorisedFeatures` prop had me quite confused 🙂.

> If this is not necessary, then I suggest we move this back to the way it was before and leave for consideration when the more holistic refactor happens. 

👍 Updated: [c12a6f3](https://github.com/Automattic/wp-calypso/pull/94044/commits/c12a6f3576377eaac9e610e3e139cea20f3cc102)

oswian commented:
> I haven't gone through the code yet, will do soon! I have some additional feedback to share:

Hey @aneeshd16, just checking to see whether you have code/other feedback for me to address?

aneeshd16 commented:
Hi @oswian, thanks for the ping! I do not have any code feedback to share.

I was wondering if the number of features listed in the comparison grid for both treatment variants is expected to be different from production. I noticed that both variants are missing the `Premium themes` feature and multiple features under the `Commerce solutions` category.

oswian committed:
Fix some feature title capitalisation

oswian committed:
Match comparison grid with production across variants

oswian commented:
> I was wondering if the number of features listed in the comparison grid for both treatment variants is expected to be different from production. I noticed that both variants are missing the Premium themes feature and multiple features under the Commerce solutions category.

@aneeshd16 Thanks for spotting this -- the number of features should match production. Update pushed.

aneeshd16 commented:
> @aneeshd16 Thanks for spotting this -- the number of features should match production. Update pushed.

Thanks for fixing it! The update, however, seems to have undone the change mentioned in https://github.com/Automattic/wp-calypso/pull/94044#issuecomment-2340167316. The incorrect sharing feature is visible in the comparison grid:
<img width="1323" alt="image" src="https://github.com/user-attachments/assets/5288657e-1558-4fb4-89ad-ea3f0584ddf7">

oswian commented:
> Thanks for fixing it! The update, however, seems to have undone the change mentioned in https://github.com/Automattic/wp-calypso/pull/94044#issuecomment-2340167316. The incorrect sharing feature is visible in the comparison grid:

@aneeshd16 Oh great spot, thanks! Fix pushed. I've also added two experiment stories for the comparison grid to help review/verify any further changes. 

As I'm now a bit paranoid about subtle regressions, Claude and I created a screenshot diffing tool that runs locally through each variant and highlights any changes 😅.

oswian committed:
Update vip features for treatment

oswian commented:
Update pushed for the VIP plan features in 'Treatment A': https://github.com/Automattic/wp-calypso/pull/94044/commits/2bfb9717a9ac4b1e2b82ea0fcecca666a91061ab

- Removed 'Professional Services'
- Removed 'Integrated APM'
- Added 'FedRAMP certification'

| Control | Treatment A (fix inaccuracies) | Treatment B (fewer features) |
| -- | -- | -- |
|<img width=260 src="https://github.com/user-attachments/assets/d055d655-6987-4f6f-a05a-e57462f3484a" />|<img width=260 src="https://github.com/user-attachments/assets/0efe0482-b07e-4bf8-9f1e-572846ec53f8" />|<img width=260 src="https://github.com/user-attachments/assets/a037b68f-72a7-4d8d-84c1-5e7e8165713d" />|

oswian commented:
Thanks again for the reviews. Proceeding with merge.

Notes:
- Once decisions are made for these two questions, I'll create followup PRs:
  - p5uIfZ-fIJ-p2#comment-23834
  - p5uIfZ-fIJ-p2#comment-23848
- Your review status @chriskmnds is still 'requested changes' however I don't believe there is any further feedback to address, so perhaps you forgot to dismiss.
- There appears to be a [flaky test](https://teamcity.a8c.com/viewLog.html?buildId=13335262&buildTypeId=calypso_calypso_WebApp_Calypso_E2E_Playwright_mobile) that passes locally.
